### PR TITLE
DROOLS-2237: Turn on CheckStyle for dmn editor

### PR DESCRIPTION
**Pull request by jomarko**
The checkstyle will control imports and the license of each java file

@manstis please have a look.

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/InputClauseColumnHeaderMetaData.java [hunk 1]
```diff
@@ -41,4 +41,9 @@ public boolean equals(final Object o) {
         // No implementation of equals/hashCode as each instance is considered different to another
         return this == o;
     }
+
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/InputClauseColumnHeaderMetaData.java]: Obviously this isn't strictly necessary, but I assume check-style (or some other plugin) borked on there being a ```equals(..)``` implementation without a ```hashCode()```.

Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/InputClauseColumnHeaderMetaData.java]: Yes, the check-style plugin was complaining about missing `hashCode` where `equals` was present, do you have other tip how to solve? I just called `super.harsCode()`.

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/InputClauseColumnHeaderMetaData.java]: No, that's fine (although a comment stating why it simply calls ```super.hashCode()``` would (should!) prevent people from removing it in the future.

Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/InputClauseColumnHeaderMetaData.java]: Oh, sure, no problem, good point,

**Comment by jomarko** [comment]
@manstis I mentioned reason for these default hashCode implementations.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by jomarko** [comment]
jenkins please retest this

**Comment by jomarko** [comment]
Failed test: `org.kie.workbench.common.services.refactoring.backend.server.drl.IndexDrlLHSTypeTest.testIndexDrlLHSType`

jenkins please retest this

**Comment by jomarko** [comment]
@manstis I can hardly believe, but jenkins says - no fail.

**Pull request merged**